lib/galaxy/datatypes/mothur.py

```diff
@@ -32,10 +35,11 @@ def set_meta(self, dataset, overwrite=True, **kwd):
             data_lines = 0
             comment_lines = 0
 
-            headers = get_headers(dataset.file_name, sep='\t', count=-1)
+            headers = iter_headers(dataset.file_name, sep='\t', count=-1)
+            first_line = next(headers)
             # set otulabels
-            if len(headers[0]) > 2:
-                otulabel_names = headers[0][2:]
+            if len(first_line) > 2:
+                otulabel_names = first_line[2:]
             # set label names and number of lines
             for line in headers:
```

Comment:
Here you are not processing the first line any more because of the `next()` call.